### PR TITLE
[#27] [UI] As a user, I can see the single-answer multiple choice question

### LINKED ISSUE #27: [UI] As a user, I can see the single-answer multiple choice question
## Why

The user can see the single-answer multiple choice question and select one option as the answer. The user selects the answer by tapping on the option.

## Acceptance criteria

Show all options of the multiple-choice question in a list. The chosen answer will be highlighted.

## Design

- [Design for a single-answer multiple choice question](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=34%3A765&scaling=scale-down)

### LINKED ISSUE #31: [UI] As a user, I can see the yes/no question
## Why

The user can see the yes/no question and select one option. The user chooses the answer by tapping on one answer.

## Acceptance criteria

Show a `Yes` and `No` option.

After the user chooses the answer, the chosen options will be highlighted. The unchosen will be blurred.

## Design

- [Design for a yes/no question](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=48%3A0&scaling=scale-down)

### PR DESCRIPTION
**Pull request by nmint8m**
- Close #27

## What happened 👀

The user can see the single-answer multiple-choice question and select one option as the answer. The user selects the answer by tapping on the option.

## Insight 📝

I implemented highlighting the selected option and create a dummy list of answers.

### CODE REVIEW

File: lib/views/answer/single_choice/single_choice_view_model.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:kayla_flutter_ic/views/answer/single_choice/single_choice_option_ui_model.dart';
+import 'package:kayla_flutter_ic/views/answer/single_choice/single_choice_state.dart';
+
+class SingleChoiceViewModel extends StateNotifier<SingleChoiceState> {
```
Comment (manh-t) [inline lib/views/answer/single_choice/single_choice_view_model.dart]: Should we create a ViewModel or State for a type of question? If we do this way, I think we have to create multiple view models / states, which are boilerplate files but bring no benefit 🤔 
Can we only have a view for a type of question, and submit the answer to `question view model` instead?

File: lib/views/answer/single_choice/single_choice_state.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:kayla_flutter_ic/views/answer/single_choice/single_choice_option_ui_model.dart';
+
+part 'single_choice_state.freezed.dart';
+
+@freezed
+class SingleChoiceState with _$SingleChoiceState {
```
Comment (manh-t) [inline lib/views/answer/single_choice/single_choice_state.dart]: I think we don't need state too

File: lib/views/answer/single_choice/single_choice_view_model.dart (hunk 1 above)
Comment (nmint8m) [reply lib/views/answer/single_choice/single_choice_view_model.dart]: From my perspective, when the user selects an option, that option will be highlighted. 

This means when the user selects an option, the state will be changed, and the view will be rebuilt.

Comment (nmint8m) [reply lib/views/answer/single_choice/single_choice_view_model.dart]: Please take a look at [this draft PR](https://github.com/nmint8m/flutter-survey-app/pull/95), I plan to use that approach.

**Comment by doannimble** [comment]
@nmint8m Please resolve the conflict 🙏

**Comment by nmint8m** [comment]
@manh-t @doannimble I resolved the conflicts, please check the PR again.

File: lib/views/answer/single_choice/single_choice_view.dart [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+      child: Container(
+        padding: const EdgeInsets.all(20),
+        child: _listView,
+      ),
+    );
+  }
+
+  void _setUpData() {
+    Future.delayed(
+      Durations.fiftyMillisecond,
```
Comment (doannimble) [inline lib/views/answer/single_choice/single_choice_view.dart]:
```suggestion
      Durations.zero,
```

**Review by doannimble — APPROVED** [review]
LGTM

File: lib/views/answer/single_choice/single_choice_view_model.dart (hunk 1 above)
Comment (nmint8m) [reply lib/views/answer/single_choice/single_choice_view_model.dart]: @manh-t 

I agree now. I was overthinking about managing the state. 

I simplified the implementation to your suggestion. Just removed the ViewModel and State of the `SingleChoiceView`. Please take a look.

File: lib/views/answer/single_choice/single_choice_state.dart (hunk 2 above)
Comment (nmint8m) [reply lib/views/answer/single_choice/single_choice_state.dart]: Sure, removed it already.

**Review by manh-t — APPROVED** [review]
LGTM 🚀

**Pull request merged**